### DISCUSSION TITLE
Not So "New" Ideas

### DISCUSSION THREAD

**Discussion by Nooby-001** [psoukie/zipchord#139] [Ideas]
I was thinking about software I've used before and though of features that maybe Zipchord could have. I'm sure everybody knows them but here it is:

1- Autocomplete V2: Start typing "the" and a window asks you if you want "them" ''their" "they're"
Pros: Could speed up typing 
Cons: 
    •Could conflict with the existing autocomplete
          •I never liked the way other apps implement it

2- Auto-correct: You type "acidentaly" and Zipchord changes it to "accidentally"
I know I can set a chord or an autocomplete for that word but frankly I don't want to memorize hundreds of "combos" 
My plan, so far, is to just set the most used words to chords (where did I read that? 😄) and maybe a hundred words as autocomplete.
It would need a big dictionary tho.

3- Restricted to... setting
Like I've mention before, I (and many other non-techies) are weary of apps that monitor keystrokes, I think a setting that deactivates Zipchord when typing on a specific app (password managers, bank emails) could make it more trustworthy.
I don't remember which app had a thing where a window would tell you "XXXX is deactivated for Firefox" and then "XXXX is activated for Notepad" that would be almost perfect.

Hope this helps

**Comment by psoukie** [comment]
Thank you for the suggestions. Quick comments

> 1- Autocomplete V2: Start typing "the" and a window asks you if you want "them" ''their" "they're"

This is indeed outside of ZipChord's philosophy: user shouldn't need to look at what is being typed, and it should help the user write faster what they want to say instead of making predictions.


> 2- Auto-correct: You type "acidentaly" and Zipchord changes it to "accidentally"

There's a lot of tools and spellcheckers out there. As you write, technically, anyone could create a shorthands dictionary for ZipChord that does this. Obviously, that would be a hugely sub-standard way of doing this. Right now, it's not something I'm considering. (Language support is another consideration here.)

> 3- Restricted to... setting
> Like I've mention before, I (and many other non-techies) are weary of apps that monitor keystrokes, I think a setting that deactivates Zipchord when typing on a specific app (password managers, bank emails) could make it more trustworthy.

Version 2.1 includes a pause feature that can be assigned any keyboard shortcut. When the Pause feature is in effect, there is a confirmation shown in OSD. As part of this, ZipChord completely releases tracking of the keys and (with the exception of the customized keyboard shortcuts -- if any are defined) it cannot know what you are typing or doing. 

I am considering extending this to where there is an automatic or semiautomatic switching of the configurations based on the active window, along the lines you suggest.
